src/apis/web3api.py

```diff
@@ -0,0 +1,213 @@
+"""
+Web3API for fetching relevant data using the web3 library.
+"""
+# pylint: disable=logging-fstring-interpolation
+
+from os import getenv
+from typing import Any, Optional
+from fractions import Fraction
+from dotenv import load_dotenv
+from web3 import Web3
+from web3.types import TxData, TxReceipt
+from eth_typing import Address, HexStr
+from hexbytes import HexBytes
+from contracts.gpv2_settlement import gpv2_settlement
+from src.models import Trade, OrderData, OrderExecution
+from src.helper_functions import get_logger
+from src.constants import SETTLEMENT_CONTRACT_ADDRESS
+
+
+class Web3API:
+    """
+    Class for fetching data from a Web3 API.
+    """
+
+    def __init__(self):
+        load_dotenv()
+        infura_key = getenv("INFURA_KEY")
+        self.url = f"https://mainnet.infura.io/v3/{infura_key}"
+        self.web_3 = Web3(Web3.HTTPProvider(self.url))
+        self.contract = self.web_3.eth.contract(
+            address=Address(HexBytes(SETTLEMENT_CONTRACT_ADDRESS)), abi=gpv2_settlement
+        )
+        self.logger = get_logger()
+
+    def get_current_block_number(self) -> Optional[int]:
+        """
+        Function that returns the current block number
+        """
+        try:
+            return int(self.web_3.eth.block_number)
+        except ValueError as err:
+            self.logger.warning(f"Error while fetching block number: {err}")
+            return None
+
+    def get_tx_hashes_by_block(self, start_block: int, end_block: int) -> list[str]:
```

Comment:
I see some inconsistency here. Other methods, when they raise an exception, return `None`. This seems to return the empty list, which makes it impossible to distinguish between cases where there is indeed an exception and cases where there is no settlement within a block range.

Reply:
I will change it to conform to our convention.

Reply:
Fixed.